src/main/drivers/accgyro_mpu.c

```diff
@@ -234,8 +234,7 @@ static bool detectSPISensorsAndUpdateDetectionResult(gyroDev_t *gyro, const sens
 #else
     UNUSED(bus);
 #endif
-    if (mpu6000SpiDetect(&gyro->bus)) {
-        gyro->mpuDetectionResult.sensor = MPU_60x0_SPI;
+    if ((gyro->mpuDetectionResult.sensor = mpu6000SpiDetect(&gyro->bus))) {
```

Comment:
Yes, I've seen that. It's in `gyroDetect` which is called after `mpuDetect`, so the `dev->mpuDetectionResult.sensor` value will be set at this point.